Bug: TestRelationSetAllTypes is flakey

### Bug description

Occasionally running the tests for the Postgres connector results in an interface conversion panic. This is likely due to pglogrepl messages being received either out of order or at unexpected times.
```
=== RUN   TestRelationSetAllTypes
--- FAIL: TestRelationSetAllTypes (0.14s)
panic: interface conversion: pglogrepl.Message is *pglogrepl.CommitMessage, not *pglogrepl.RelationMessage [recovered]
        panic: interface conversion: pglogrepl.Message is *pglogrepl.CommitMessage, not *pglogrepl.RelationMessage

goroutine 11 [running]:
testing.tRunner.func1.2({0x1a41de0, 0xc00033c570})
        /usr/local/go/src/testing/testing.go:1389 +0x366
testing.tRunner.func1()
        /usr/local/go/src/testing/testing.go:1392 +0x5d2
panic({0x1a41de0, 0xc00033c570})
        /usr/local/go/src/runtime/panic.go:844 +0x258
github.com/conduitio/conduit-connector-postgres/source/logrepl/internal.TestRelationSetAllTypes(0xc0000036c0)
        /Users/dylanlott/dev/meroxa/conduit-connector-postgres/source/logrepl/internal/relationset_test.go:58 +0x61b
testing.tRunner(0xc0000036c0, 0x1b48c08)
        /usr/local/go/src/testing/testing.go:1439 +0x214
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:1486 +0x725
FAIL    github.com/conduitio/conduit-connector-postgres/source/logrepl/internal 1.480s
```

### Steps to reproduce

1. Run `make run`
2. The `TestRelationSetAllTypes` test occasionally fails with a failed interface type conversion.

### Version

v0.1.0

--- a/source/logrepl/internal/relationset_test.go
+++ b/source/logrepl/internal/relationset_test.go
@@ -61,9 +61,7 @@ func TestRelationSetAllTypes(t *testing.T) {
 	for {
 		// first message needs to be a begin message
 		msg := <-messages
-		if _, ok := msg.(*pglogrepl.BeginMessage); !ok {
-			t.Fatalf("expected begin message, got %s", msg.Type())
-		}
+		_ = msg.(*pglogrepl.BeginMessage)
 
 		// second message can be either commit (we can catch empty transactions)
 		// or relation (that's what we are actually interested in)
@@ -75,13 +73,7 @@ func TestRelationSetAllTypes(t *testing.T) {
 		// not an empty transaction, these have to be the messages we are looking for
 		rel = msg.(*pglogrepl.RelationMessage)        // second message is a relation
 		ins = (<-messages).(*pglogrepl.InsertMessage) // third one is the insert
-
-		// last one needs to be commit
-		msg = <-messages
-		if _, ok := msg.(*pglogrepl.CommitMessage); !ok {
-			t.Fatalf("expected commit message, got %s", msg.Type())
-		}
-
+		_ = (<-messages).(*pglogrepl.CommitMessage)   // fourth one is the commit
 		break
 	}
 

--- a/source/logrepl/internal/relationset_test.go
+++ b/source/logrepl/internal/relationset_test.go
@@ -54,10 +54,36 @@ func TestRelationSetAllTypes(t *testing.T) {
 	_, messages := setupSubscription(ctx, t, conn.Config().Config, table)
 	insertRowAllTypes(ctx, t, conn, table)
 
-	<-messages                                       // skip first message, it's a begin message
-	rel := (<-messages).(*pglogrepl.RelationMessage) // second message is a relation
-	ins := (<-messages).(*pglogrepl.InsertMessage)   // third one is the insert
-	<-messages                                       // fourth one is the commit
+	var rel *pglogrepl.RelationMessage
+	var ins *pglogrepl.InsertMessage
+
+	// loop through messages and allow for empty transactions to be received
+	for {
+		// first message needs to be a begin message
+		msg := <-messages
+		if _, ok := msg.(*pglogrepl.BeginMessage); !ok {
+			t.Fatalf("expected begin message, got %s", msg.Type())
+		}
+
+		// second message can be either commit (we can catch empty transactions)
+		// or relation (that's what we are actually interested in)
+		msg = <-messages
+		if _, ok := msg.(*pglogrepl.CommitMessage); ok {
+			continue // empty transaction, skip it
+		}
+
+		// not an empty transaction, these have to be the messages we are looking for
+		rel = msg.(*pglogrepl.RelationMessage)        // second message is a relation
+		ins = (<-messages).(*pglogrepl.InsertMessage) // third one is the insert
+
+		// last one needs to be commit
+		msg = <-messages
+		if _, ok := msg.(*pglogrepl.CommitMessage); !ok {
+			t.Fatalf("expected commit message, got %s", msg.Type())
+		}
+
+		break
+	}
 
 	rs := NewRelationSet(conn.ConnInfo())
 

